.changeset/metal-ties-prove.md

```diff
@@ -0,0 +1,22 @@
+---
+'xstate': major
+---
+
+Custom action objects and guard objects are now expected to put extra parameters on the `params` property:
```

Comment:
It's much better for types, because `{ type: string; [key: string]: any; exec?: ... }` caused problems.

So I think for the TypeScript DX alone it's worth it. Also, now the objects have a predictable shape.